adobe/spectrum-css
# chore(popover): update popover stories

## Description

- This updates the Popover stories to include an Action Button as the trigger source for the Popover to allow for a better visual. 
- Updates the Spectrum Website link for popover on the docs site from `popovers` to `popover`
- Adds an onclick event to activate the popover for the story
- Updates a few other components to pass through functions for content to account for an error that the Interaction add-on was causing (IllustratedMessage, Dialog, Card)

## To test

1. Run Storybook for this PR and navigate to the Popover story
2. Click the button to open the Popover
3. Change the position of the Popover and make sure that it updates accordingly (you will have to close and reopen the popover as the positioning is done with javascript on button click)

## To-do list



- [x] I have updated any relevant storybook stories and templates.
